ext/nnstreamer/tensor_filter/tensor_filter_openvino.cc

```diff
@@ -0,0 +1,421 @@
+/**
+ * GStreamer Tensor_Filter, OpenVino (DLDT) Module
+ * Copyright (C) 2019 Wook Song <wook16.song@samsung.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Library General Public
+ * License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Library General Public License for more details.
+ *
+ */
+/**
+ * @file    tensor_filter_openvino.cpp
+ * @date    23 Dec 2019
+ * @brief   Tensor_filter subplugin for OpenVino (DLDT).
+ * @see     http://github.com/nnsuite/nnstreamer
+ * @author  Wook Song <wook16.song@samsung.com>
+ * @bug     No known bugs except for NYI items
+ *
+ * This is the per-NN-framework plugin (OpenVino) for tensor_filter.
+ */
+
+#include <glib.h>
+#include <nnstreamer_plugin_api_filter.h>
+#include <tensor_common.h>
+
+#include <inference_engine.hpp>
+#include <iostream>
+#include <string>
+#include <vector>
+
+void init_filter_openvino (void) __attribute__ ((constructor));
+void fini_filter_openvino (void) __attribute__ ((destructor));
+
+class TensorFilterOpenvino
+{
+public:
+  enum RetVal
+  {
+    RetSuccess = 0,
+    RetEInval = -EINVAL,
+    RetEOverFlow = -EOVERFLOW,
+  };
+
+  static tensor_type convertFromIETypeStr (std::string type);
+
+  TensorFilterOpenvino (std::string path_model_prefix);
+  ~TensorFilterOpenvino ();
+
+  int getInputTensorDim (GstTensorsInfo * info);
+  int getOutputTensorDim (GstTensorsInfo * info);
+  std::string getPathModelXml ();
+  std::string getPathModelBin ();
+
+  static const std::string extBin;
+  static const std::string extXml;
+
+private:
+  TensorFilterOpenvino ();
+
+  InferenceEngine::Core _ieCore;
+  InferenceEngine::CNNNetReader _networkReaderCNN;
+  InferenceEngine::CNNNetwork _networkCNN;
+  InferenceEngine::InputsDataMap _inputsDataMap;
+  InferenceEngine::TensorDesc _inputTensorDesc;
+  InferenceEngine::OutputsDataMap _outputsDataMap;
+
+  std::string pathModelXml;
+  std::string pathModelBin;
+};
+
+const std::string TensorFilterOpenvino::extBin = ".bin";
+const std::string TensorFilterOpenvino::extXml = ".xml";
+
+/**
+ * @brief Convert the string representing the tensor data type to _nns_tensor_type
+ * @param type a std::string representing the tensor data type in InferenceEngine
+ * @return _nns_tensor_type corresponding to the tensor data type in InferenceEngine if OK, otherwise _NNS_END
+ */
+tensor_type
+TensorFilterOpenvino::convertFromIETypeStr (std::string type)
+{
+  if (type[0] == 'U') {
```

Comment:
I think it's better to compare the exact string unless `type` has a random suffix, for correct error handling of type conversion.

Reply:
Resolved in offline :)